api/src/main/java/edu/cornell/mannlib/vitro/webapp/controller/freemarker/IndividualListController.java

```diff
@@ -228,9 +234,12 @@ private static IndividualListQueryResults buildAndExecuteVClassQuery(
 	}
 
     private static IndividualListQueryResults buildAndExecuteRandomVClassQuery(
-			List<String> vclassURIs, int page, int pageSize, IndividualDao indDao)
+			List<String> vclassURIs, int page, int pageSize, VitroRequest vreq)
 			throws SearchEngineException {
+		 IndividualDao indDao = vreq.getWebappDaoFactory().getIndividualDao();
 		 SearchQuery query = SearchQueryUtils.getRandomQuery(vclassURIs, page, pageSize);
+		 Set<String> currentRoles = SearchFiltering.getCurrentUserRoles(vreq);
```

Comment:
The same comment as the last one

Reply:
I think it is not used in loops, so I think performance should be ok.